lms/static/scripts/frontend_apps/components/dashboard/test/AssignmentActivity-test.js

```diff
@@ -561,6 +575,106 @@ describe('AssignmentActivity', () => {
         })),
       });
     });
+
+    [
+      {
+        fetchResult: { data: null },
+        expectedResult: false,
+      },
+      {
+        fetchResult: {
+          data: { status: 'scheduled' },
+        },
+        expectedResult: true,
+      },
+      {
+        fetchResult: {
+          data: { status: 'in_progress' },
+        },
+        expectedResult: true,
+      },
+      {
+        fetchResult: {
+          data: { status: 'finished' },
+        },
+        expectedResult: false,
+      },
+    ].forEach(({ fetchResult, expectedResult }) => {
+      it('shouldRefresh callback behaves as expected', () => {
+        createComponent();
+        assert.equal(shouldRefreshCallback(fetchResult), expectedResult);
+      });
+    });
+
+    [
+      {
+        data: null,
+        shouldDisplayLastSyncDate: false,
```

Comment:
```suggestion
        shouldDisplayLastSyncInfo: false,
```

This field refers to whether the "Grades last synced" line is shown, but that may not include a _date_ (such as when the first sync is in progress).